src/psyclone/tests/psyir/symbols/symboltable_test.py

```diff
@@ -1486,54 +1486,95 @@ def test_new_symbol():
             " one of its sub-classes but found" in str(err.value))
 
 
-def test_symbol_from_tag():
-    ''' Tests the SymbolTable symbol_from_tag method '''
+def test_find_or_create():
+    ''' Tests the SymbolTable find_or_create method find existing symbols or
+    otherwise creates a new symbol with the given properties. '''
+    symtab = SymbolTable()
+    existing_symbol = Symbol("existing")
+    symtab.add(existing_symbol, tag="tag1")
+
+    # If the given name exists, return the symbol
+    assert symtab.find_or_create("existing") is existing_symbol
+
+    # If the given name does not exist, crate and return new symbol
+    new1 = symtab.find_or_create("new1")
+    assert isinstance(new1, Symbol)
+    assert new1.name == "new1"
+    assert new1 is symtab.find_or_create("new1")  # Which then is found
+
+    # Creating symbols can have parameters passed to the new_symbol method
+    new2 = symtab.find_or_create("new2",
+                                 tag="mytag",
+                                 symbol_type=DataSymbol,
+                                 datatype=INTEGER_TYPE,
+                                 visibility=Symbol.Visibility.PRIVATE,
+                                 constant_value=3)
+    assert new2.name == "new2"
+    assert isinstance(new2, DataSymbol)
+    assert new2.datatype is INTEGER_TYPE
+    assert new2.visibility is Symbol.Visibility.PRIVATE
+    assert new2.constant_value.value == "3"
+    assert symtab.lookup_with_tag("mytag") is new2
+
+    # Check that it fails if the Symbol type is different than expected by
```

Comment:
"...if the named Symbol exists but is not of the specified type."

Reply:
Fixed